modules/dataframe/src/util/dataframeutil.cpp

```diff
@@ -524,6 +524,64 @@ std::shared_ptr<DataFrame> combineDataFrames(std::vector<std::shared_ptr<DataFra
     return newDataFrame;
 }
 
+#include <warn/push>
+#include <warn/ignore/conversion>
+std::vector<std::uint32_t> filteredRows(const Column& col, dataframefilters::Filters filters) {
+
+    if (col.getColumnType() == ColumnType::Categorical) {
+        const auto& catCol = dynamic_cast<const CategoricalColumn&>(col);
+        std::vector<std::uint32_t> rows;
+        for (auto&& [row, value] : util::enumerate<std::uint32_t>(catCol.getValues())) {
+            auto test = util::overloaded{
+                [v = value](const std::function<bool(std::string_view)>& func) { return func(v); },
+                [](const auto&) { return false; }};
```

Comment:
Might consider adding the different signatures explicitly here and below, to make sure you get and error if the variant types change...

Reply:
also should this not return true, when in combination with the any_of below, I would expect a int filter in a categorical column to fail. not be ignored in the any_of?